onmt/dynamic/opts.py

```diff
@@ -0,0 +1,96 @@
+"""All options for dynamic running. Should be given in a yaml file."""
+from onmt.opts import _train_general_opts, config_opts, model_opts
+from onmt.dynamic.transforms import AVAILABLE_TRANSFORMS
+
+
+def data_config_opts(parser):
+    """Options related to dynamic data config file."""
+    parser.add('-data_config', '--data_config', required=True,
+               is_config_file_arg=True, help='data config file path')
+
+
+def _dynamic_corpus_opts(parser):
+    """Options related to training corpus, type: a list of dictionary."""
+    group = parser.add_argument_group('Data')
+    group.add('-data', '--data', required=True,
+              help='data configuration list.')
+    group.add('-save_data', '--save_data', required=True,
+              help='Output directory for data saving path.')
+    group.add('-overwrite', '--overwrite', action="store_true",
+              help="Overwrite existing shards if any.")
+    group.add('-transforms', '--transforms', default=[], nargs='+',
+              choices=AVAILABLE_TRANSFORMS.keys(),
+              help="Default transform pipeline applying to data."
+                   "Can be specified in each corpus of data to override.")
+
+
+def _dynamic_vocab_opts(parser):
+    """Options related to vocabulary and fields."""
+    group = parser.add_argument_group('vocab')
+    group.add('-src_vocab', '--src_vocab', required=True,
```

Comment:
This means -src_vocab is required in `build_vocab` execution. Not very straightforward... :)

Reply:
Addressed in #18.